library/core/src/slice/mod.rs

```diff
@@ -3293,8 +3293,9 @@ mod private_slice_index {
     on(T = "str", label = "string indices are ranges of `usize`",),
     on(
         all(any(T = "str", T = "&str", T = "std::string::String"), _Self = "{integer}"),
-        note = "you can use `.chars().nth()` or `.bytes().nth()`
-see chapter in The Book <https://doc.rust-lang.org/book/ch08-02-strings.html#indexing-into-strings>"
+        note = "you can use `.chars().nth()` or `.bytes().nth()`\n\
+                For more information, see chapter 8 in The Book: \
```

Comment:
Second nit: this should be lowercase for consistency.
```suggestion
                for more information, see chapter 8 in The Book: \
```